devito/finite_differences/derivative.py

```diff
@@ -192,22 +199,39 @@ def T(self):
             adjoint = direct
 
         return Derivative(self.expr, *self.dims, deriv_order=self.deriv_order,
-                          fd_order=self.fd_order, side=self.side, stagger=self.stagger,
-                          transpose=adjoint)
+                          fd_order=self.fd_order, side=self.side, transpose=adjoint,
+                          x0=self.x0, subs=self._subs)
+
+    def _eval_at(self, func):
```

Comment:
Is there a good reason this can't/shouldn't be called from evaluate (if certain properties etc. are present)?

Reply:
This is still within lazy evaluation, this doesn't  do anything except setting x0 so I prefer it separated yes.

Reply:
Yeah, after having time to think about it a little more I'm fully on board with the current implementation.